reactions/src/HALO.cpp

```diff
@@ -671,6 +715,147 @@ double HALO::plinear_cosmosis(double k) const {
 }
 
 
+// vars:
+// 0 = acol, 1= omega0, 2 = mg param
+void HALO::react_init_nu(double vars[]) const{
+  SPT spt(C, P_l, epsrel);
+
+  double pspt, psptp, p1h, p1hp, plm, plcb, plnu, psptcbv, argument;
+
+  // fv
+  double fv  = vars[1]/vars[7];
+  double fv2 = pow(fv,2);
+  double fvt = 1.-fv;
+  double fvt2  = pow(fvt,2);
+
+/// mathcal E///
+  bigE = pow2(1.-fv)*one_halo(0.01,vars)/one_halop(0.01,vars);
+
+  /// Calculate kstar///
+// linear terms
+  plm = P_l(0.06);
+  plcb = P_cb(0.06);
+  plnu = P_nu(0.06);
+
+// 1halo terms
+  p1h = one_halo(0.06,vars); // real
+  p1hp = one_halop(0.06,vars); // pseudo
+
+// spt terms
+// Real PT spectrum
+  pspt = spt.PLOOPn2(1, vars, 0.06, 1e-3) + p1h ;
+// GR-1-loop spectrum with linear growth replaced by modified gravity growth (unscreened)
+  psptp = spt.PLOOPn2(8, vars, 0.06, 1e-3)+ p1hp ;
+
+// cbv terms
+  psptcbv = pspt*plnu;
+
+// reactionm @ k0 with 2halo term = 1 loop spt
+  double rsptk0 = (fvt2 * pspt + 2.*fv*fvt*sqrt(psptcbv) + fv2*plnu)/(psptp);
+
+
+  double prefac = 1./((bigE-1.)*plcb*fvt2);
+  double term1 = p1h*fvt2 + (fvt2*bigE*plcb - fv2*plnu - (p1hp+plm)*rsptk0);
+  double term2 = 2.*sqrt(fvt2*fv2*(p1hp+plm)*plnu*rsptk0);
+
+  argument = prefac * (term1 + term2); // 1st root
+// argument = prefac * (term1 - term2); // 2nd root
+
+  if(argument<0.001 || argument>1.){
+    kstar = 1e-6;
+    bigE = 1.;
+  }
+  else{
+  kstar = (-0.06/log(argument));
+  }
+
+}
+
+
+// Reaction using 1-loop splines (ploopr and ploopp) as a function of redshift -- these should be initialised of course, one can use the ploop_init function in SPT.cpp
+void HALO::react_init_nu2(double vars[], Spline ploopr, Spline ploopp) const{
+  SPT spt(C, P_l, epsrel);
+
+  double pspt, psptp, p1h, p1hp, plm, plcb, plnu, psptcbv, argument;
+
+  // fv
+  double fv  = vars[1]/vars[7];
+  double fv2 = pow(fv,2);
+  double fvt = 1.-fv;
+  double fvt2  = pow(fvt,2);
+
+/// mathcal E///
+  bigE = pow2(1.-fv)*one_halo(0.01,vars)/one_halop(0.01,vars);
+
+  /// Calculate kstar///
+// linear terms
+  plm = P_l(0.06);
+  plcb = P_cb(0.06);
+  plnu = P_nu(0.06);
+
+// 1halo terms
+  p1h = one_halo(0.06,vars); // real
+  p1hp = one_halop(0.06,vars); // pseudo
+
+// spt terms
+// Real PT spectrum
+  pspt = ploopr(vars[0]) + p1h ;
+// GR-1-loop spectrum with linear growth replaced by modified gravity growth (unscreened)
+  psptp = ploopp(vars[0]) + p1hp ;
+
+// cbv terms
+  psptcbv = pspt*plnu;
+
+// reactionm @ k0 with 2halo term = 1 loop spt
+  double rsptk0 = (fvt2 * pspt + 2.*fv*fvt*sqrt(psptcbv) + fv2*plnu)/(psptp);
+
+
+  double prefac = 1./((bigE-1.)*plcb*fvt2);
+  double term1 = p1h*fvt2 + (fvt2*bigE*plcb - fv2*plnu - (p1hp+plm)*rsptk0);
+  double term2 = 2.*sqrt(fvt2*fv2*(p1hp+plm)*plnu*rsptk0);
+
+  argument = prefac * (term1 + term2); // 1st root
+// argument = prefac * (term1 - term2); // 2nd root
+
+  if(argument<0.001 || argument>1.){
+    kstar = 1e-6;
+    bigE = 1.;
+  }
+  else{
+  kstar = (-0.06/log(argument));
```

Comment:
would be good to reference Equation (6) in Bose et al here (and in general it will improve user friendliness if important equations are referenced with arXiv number included) -- again, would be much better if instead of 0.06 we had k_bar

Reply:
Added reference.